pom.xml

```diff
@@ -322,11 +325,41 @@
                 <arg>${cmake.generated.directory}</arg>
               </compilerArgs>
             </configuration>
+            <executions>
+              <!-- compile everything to ensure module-info contains right entries -->
+              <execution>
+                <id>default-compile</id>
+                <configuration>
+                  <release>9</release>
+                  <compileSourceRoots>
+                    <sourceRoot>src/main/java</sourceRoot>
+                    <sourceRoot>src/main/module-info</sourceRoot>
+                  </compileSourceRoots>
+                </configuration>
+              </execution>
+              <!-- recompile everything for target VM except the module-info.java -->
+              <execution>
+                <id>minimal-target-jre-recompile</id>
+                <goals><goal>compile</goal></goals>
+              </execution>
+            </executions>
           </plugin>
         </plugins>
       </build>
     </profile>
 
+    <!-- Convenience profile for users of JDKs past 11 who may want to
+         build the library ignoring no longer supported target version 6 -->
+    <profile>
+      <id>jdk12+</id>
+      <activation>
+        <jdk>[12,)</jdk>
+      </activation>
+      <properties>
+        <target.java.version>7</target.java.version>
```

Comment:
So just to double check before merging, this will automatically activate and supersede the previous property when run on 12+, right?

Reply:
That's right. Properties inside profiles overwrite the global ones, and the profiles add. Check for different jdks:
```
$ JAVA_HOME=/home/me/.jdks/corretto-15.0.2/ mvn clean install  help:active-profiles 
$ JAVA_HOME=/home/me/.jdks/adopt-openjdk-11.0.10/ mvn clean install  help:active-profiles 
etc
```